src/Process/Deploy/PreDeploy/CheckEnvFile.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Process\Deploy\PreDeploy;
+
+use Magento\MagentoCloud\App\GenericException;
+use Magento\MagentoCloud\Config\Deploy\Reader;
+use Magento\MagentoCloud\Config\Deploy\Writer;
+use Magento\MagentoCloud\Config\State;
+use Magento\MagentoCloud\Config\Validator\Deploy\DatabaseConfiguration;
+use Magento\MagentoCloud\Config\Validator\Result\Error;
+use Magento\MagentoCloud\Filesystem\BackupList;
+use Magento\MagentoCloud\Filesystem\Driver\File;
+use Magento\MagentoCloud\Filesystem\FileList;
+use Magento\MagentoCloud\Process\ProcessException;
+use Magento\MagentoCloud\Process\ProcessInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Checks if app/etc/env.php exists and restores it if not.
+ */
+class CheckEnvFile implements ProcessInterface
```

Comment:
It's better to name it `RestoreEnvFile`

Reply:
I wanted to name `RestoreEnvFile` too, but this process does not always restore env file but do some other stuff.
`Restore` - doesn't describe full responsibility of the class. 
P.S. I'm not saying that `Check` is a good name.